supporting_components/graph.py

```diff
@@ -440,18 +440,21 @@ def copy(self):
         copy = Graph(self.directed)
         copy.max_colornum = self.max_colornum
         vertices_old_to_new = {}
+        colors = {}
```

Comment:
Bestaat er alleen een colors dictionary als de Graph gekopieerd is?

Reply:
`colors` wordt geïnitialiseerd bij de `color_initialization` methode. Bij de `copy` methode wordt een hele nieuwe dict gevuld, omdat het met `dict(colors)` en `colors.copy()` niet werkte. Wanneer ik dan wat veranderde aan `colors`, veranderde dat ook in `colors_backup`. Nu is `colors_backup` een nieuwe dict met de stand van zaken op dat moment en wordt deze niet veranderd wanneer `colors` wordt veranderd.

Reply:
Ja logisch ook, anders heb je toch geen kleuren.